controllers/storagecluster/storageconsumer.go

```diff
@@ -144,9 +145,8 @@ func getLocalStorageClassNames(ctx context.Context, kubeClient client.Client, st
 		return nil, err
 	}
 	for idx := range storageClassesInCluster.Items {
-		// TODO: skip storageclasses that are from external mode if both internal & external mode is enabled
 		sc := &storageClassesInCluster.Items[idx]
-		if slices.Contains(supportedCsiDrivers, sc.Provisioner) {
+		if slices.Contains(supportedCsiDrivers, sc.Provisioner) && sc.Labels[server.ExternalClassLabelKey] != "true" {
```

Comment:
There is no point in filtering on a able when you can just list to exclude anything with the label